AlfrescoKit/AlfrescoKit/View Controllers/Browsing View Controller/AKAlfrescoNodeListViewController.m

```diff
@@ -72,20 +78,26 @@ - (void)viewDidLoad
     else
     {
         __weak typeof(self) weakSelf = self;
-        self.currentRequest = [self retrieveRootFolderWithCompletionBlock:^(AlfrescoFolder *rootFolder, NSError *rootError) {
-            if (rootError)
-            {
-                if ([weakSelf.delegate respondsToSelector:@selector(listViewController:didFailToRetrieveItemsWithError:)])
+        __block AlfrescoRequest *request = nil;
+        request = [self retrieveRootFolderWithCompletionBlock:^(AlfrescoFolder *rootFolder, NSError *rootError) {
+            [weakSelf.delegate controller:weakSelf didCompleteRequest:request error:rootError];
+            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```

Comment:
Could you add comments on why this delay is required? I can't see a reason just from inspection.